doc/helpers.md

```diff
@@ -0,0 +1,153 @@
+# Helpers
+
+This document provides a full introduction and usage example of each helper implemented here. Let's start with it!
+
+## state-flow
+Integration tests are really important in our development cycle, so this helper aims to help you with [state-flow](https://github.com/nubank/state-flow), an integration testing framework using a state monad in the backend for building and composing flows. You can write multiple flows to build your integration test. By default there are some helpers for `state-flow`, so let's deep dive into each one!
+
+### server/pedestal
+The goal of this helper is to extract `io.pedestal.http/service-fn` from `state-flow` context and call `io.pedestal.test/response-for` to simulate HTTP requests.
+
+This helper implements a `request!` function that receives a map with `method`, `uri`, `body` and `headers` (same for composing a request in pedestal) and starts a flow that *makes* an HTTP request and parses the response. You can see an example of this example [here](https://github.com/parenthesin/components/blob/main/test/integration/parenthesin/schema/system_test.clj#L79) and the context down below:
+```clojure
+;; inside your test
+;; don't forget to import corretly the libraries
+(flow "should match the response"
+  (match? {:status 200
+           :body "yay!"}
+          (state-flow.server/request! {:method :post
+                                       :uri    "/some/url"
+                                       :body   {:something "yay"}})))
+```
+> This example also uses the [matchers-combinators](https://github.com/nubank/matcher-combinators/) library to validate integration tests.
+
+By default this implementation is for `request!` it's fully prepared to do all HTTP client functionalities, like handling multipart uploads. An example of this implementation can be seen below:
```

Comment:
it's -> isn't and bellow instead the routes, I would put the definition of the request! wrapper for clj-http.client/request